### PR TITLE
Update dependencies and CI

### LINKED ISSUE #2056: Build failing in dependency installation step
### Steps to reproduce

Download vtk.js and attempt to install dependencies:

```
git clone https://github.com/Kitware/vtk-js.git
cd vtk-js
npm install
```

### Detailed behavior

Dependency installation process attempts to compile the package "canvas" and then fails with some errors.

Here's the start of it:
```
$ npm install
npm WARN old lockfile 
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile 
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile 
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated har-validator@5.1.3: this library is no longer supported
npm WARN deprecated debug@4.2.0: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated debug@3.2.6: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated eslint-loader@4.0.2: This loader has been deprecated. Please use eslint-webpack-plugin
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated node-pre-gyp@0.11.0: Please upgrade to @mapbox/node-pre-gyp: the non-scoped node-pre-gyp package is deprecated and only the @mapbox scoped package will recieve updates in the future
npm WARN deprecated node-pre-gyp@0.17.0: Please upgrade to @mapbox/node-pre-gyp: the non-scoped node-pre-gyp package is deprecated and only the @mapbox scoped package will recieve updates in the future
npm WARN deprecated core-js@2.5.7: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
npm WARN deprecated core-js@2.6.11: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
npm ERR! code 1
npm ERR! path /home/ebrahim/Desktop/vtk-js/node_modules/canvas
npm ERR! command failed
npm ERR! command sh -c node-pre-gyp install --fallback-to-build
npm ERR! make: Entering directory '/home/ebrahim/Desktop/vtk-js/node_modules/canvas/build'
npm ERR!   SOLINK_MODULE(target) Release/obj.target/canvas-postbuild.node
npm ERR!   COPY Release/canvas-postbuild.node
npm ERR!   CXX(target) Release/obj.target/canvas/src/backend/Backend.o
npm ERR!   CXX(target) Release/obj.target/canvas/src/backend/ImageBackend.o
npm ERR!   CXX(target) Release/obj.target/canvas/src/backend/PdfBackend.o
[more error messages]
```

Log file with the full list of errors:
[2021-09-15T19_49_35_798Z-debug.log](https://github.com/Kitware/vtk-js/files/7172859/2021-09-15T19_49_35_798Z-debug.log)

### Environment
- **vtk.js**: master branch, which is currently commit c6f82adb85
- **OS**: Ubuntu 20.04.2 LTS

### PR DESCRIPTION
**Pull request by floryst**
### PR and Code Checklist

- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Context

Upgrades dependencies, replaces resemble.js (canvas dependency is annoying), and updates CI node versions. Context: #2056 

### Changes

- [x] Package upgrades
- [x] Fixes failing tests
- [x] prettier upgrade changes code style
- [x] update webpack config
- [x] Replace resemble.js
- [x] Update CI node versions

### Results

Should fix #2056.

### Testing

- [x] This change adds or fixes unit tests
- [x] All tests complete without errors on the following environment:
  - **vtk.js**: 
  - **OS**: 
  - **Browser**:

### CODE REVIEW

File: webpack.dev.js [hunk 1]
```diff
@@ -33,16 +37,13 @@ module.exports = [
...
-    ],
   }),
+  //merge(common.liteConfig, {
+  //  mode: 'development',
+  //  devtool: 'inline-source-map',
+  //  devServer: configureDevServer(port + 1),
+  //  plugins: [
+  //    new webpack.HotModuleReplacementPlugin(),
+  //  ],
+  //}),
```
Comment (floryst) [inline webpack.dev.js:48]: @jourdain why do we run two webpack configs when running `npm run build`?

Comment (jourdain) [reply webpack.dev.js:48]: To have a trimmed down version of vtk.js but now that we have ESM, I don't see the need anymore.

**Review by jourdain — APPROVED** [review]
LGTM, the reformat make it hard to grasp what are the important changes... ;-)

**Comment by floryst** [comment]
Agreed. I would recommend filtering by single commits, since I isolated all of the refactoring changes to a single commit.

**Pull request merged**